sw/airborne/modules/imu/imu.c

```diff
@@ -579,12 +579,12 @@ static void imu_gyro_raw_cb(uint8_t sender_id, uint32_t stamp, struct Int32Rates
   if(gyro->calibrated.filter) {
     struct Int32Rates data_filtered[samples];
     for(uint8_t i = 0; i < samples; i++) {
-      update_butterworth_2_low_pass_int(&gyro->filter[0], data[i].p);
-      data_filtered[i].p = get_butterworth_2_low_pass_int(&gyro->filter[0]);
-      update_butterworth_2_low_pass_int(&gyro->filter[1], data[i].q);
-      data_filtered[i].q = get_butterworth_2_low_pass_int(&gyro->filter[1]);
-      update_butterworth_2_low_pass_int(&gyro->filter[2], data[i].r);
-      data_filtered[i].r = get_butterworth_2_low_pass_int(&gyro->filter[2]);
+      update_butterworth_2_low_pass(&gyro->filter[0], data[i].p);
+      data_filtered[i].p = get_butterworth_2_low_pass(&gyro->filter[0]);
```

Comment:
side note: `update` function already returns the filtered value, so you can write `data_filtered[i].p = update_butterworth_2_low_pass(&gyro->filter[0], data[i].p);`

Reply:
Thanks updated